enmeshed/main.tf

```diff
@@ -0,0 +1,258 @@
+locals {
+  mongodb_uri = "mongodb://mongo:pass@enmeshed-db:27017/enmeshed-db"
+}
+
+resource "kubernetes_deployment" "enmeshed_deployment" {
+  metadata {
+    name      = "enmeshed-deployment"
+    namespace = var.namespace
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "enmeshed-app"
+      }
+    }
+
+    template {
+      metadata {
+        namespace = var.namespace
+        labels = {
+          app = "enmeshed-app"
+        }
+      }
+      spec {
+        container {
+          image = "ghcr.io/nmshd/connector:${var.image_tags.enmeshed}"
+          name  = "connector"
+
+          env {
+            name  = "CUSTOM_CONFIG_LOCATION"
+            value = "/config.json"
+          }
+          # TODO
+          #          resources {
+          #            limits   = {
+          #              cpu    = "1000m"
+          #              memory = "2000Mi"
+          #            }
+          #            requests = {
+          #              cpu    = "750m"
+          #              memory = "1500Mi"
+          #            }
+          #          }
+          volume_mount {
+            name       = "config"
+            mount_path = "/config.json"
+            sub_path   = "config.json"
+            read_only  = true
+          }
+        }
+
+        volume {
+          name = "config"
+
+          secret {
+            secret_name = "config.json"
+          }
+        }
+      }
+    }
+  }
+}
+
+
+resource "kubernetes_service" "enmeshed" {
+  metadata {
+    name      = "enmeshed"
+    namespace = var.namespace
+
+    labels = {
+      app = "enmeshed-app"
+    }
+  }
+
+  spec {
+    selector = {
+      app = "enmeshed-app"
+    }
+
+    type = "NodePort"
+
+    port {
+      port        = 80
+      target_port = 80
+      protocol    = "TCP"
+    }
+  }
+}
+
+resource "kubernetes_service" "mongodb" {
+  metadata {
+    name      = "mongodb"
+    namespace = var.namespace
+
+    labels = {
+      app = "mongodb-app"
+    }
+  }
+
+  spec {
+    selector = {
+      app = "mongodb-app"
+    }
+
+    type = "NodePort"
+
+    port {
+      port        = 27017
+      target_port = 27017
+      protocol    = "TCP"
+    }
+  }
+}
+
+resource "kubernetes_persistent_volume_claim" "mongodb-pv-claim" {
+  metadata {
+    name      = "mongodb-pv-claim"
+    namespace = var.namespace
+
+    labels = {
+      app = "mongodb-app"
+    }
+  }
+
+  spec {
+    access_modes = ["ReadWriteOnce"]
+
+    resources {
+      requests = {
+        storage = "5Gi"
+      }
+    }
+  }
+}
+
+resource "kubernetes_deployment" "mongodb_deployment" {
```

Comment:
Might be easier to just use a Helm chart here. See https://github.com/serlo/infrastructure-modules-shared/tree/main/redis for an example on deploying Redis via Helm.